src/swamid_plugins/site_selector/__init__.py

```diff
@@ -32,21 +36,37 @@ def process(self, context, internal_data):
             if result:
                 return super().process(context, internal_data)
 
-        # no rule was successful; abort processing the response any further
-        error_context = {
-            "message": "Failed to satisfy any rule for accessing the service.",
-            "service": service_id,
-            "rules": access_rules_for_service,
+        # no rule was successful; either redirect or abort processing the response any further
+        if not self.redirect_url:
+            error_context = {
+                "message":
+                "Failed to satisfy any rule for accessing the service.",
+                "service": service_id,
+                "rules": access_rules_for_service,
+            }
+            raise Error(error_context)
+        context = {
+            "user_id": (internal_data.attributes.get(self.primary_identifier)
+                        or ["Unknown"])[0],
+            "displayname": (internal_data.attributes.get("displayname")
+                            or ["Unknown"])[0],
+            "timestamp":
+            internal_data.auth_info.timestamp,
+            "issuer":
+            internal_data.auth_info.issuer,
+            "service":
+            service_id,
+            "session_id":
+            context.state.session_id
         }
-        raise Error(error_context)
+        query_string = urlencode(
+            {"context": base64.b64encode(json.dumps(context).encode("utf-8"))})
+        return Redirect(self.redirect_url + f'?{query_string}')
```

Comment:
use urlsafe base64 encoding:

```suggestion
        context_ser = json.dumps(payload, separators=(",", ":"))
        context_ser_bytes = context_ser.encode("utf-8")
        context_enc_bytes = base64.urlsafe_b64encode(context_ser_bytes)
        context_enc = context_enc_bytes.decode("utf-8")
        query_string = urlencode({"context": context_enc})
        return Redirect(f"{self.redirect_url}?{query_string}")
```